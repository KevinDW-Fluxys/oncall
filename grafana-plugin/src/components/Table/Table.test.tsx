import React from 'react';

import { render } from '@testing-library/react';

import Table from './Table';

describe('Table', () => {
  test('it renders pagination related data properly', () => {
    const component = render(
      <Table
        data={[{ foo: 'bar' }, { foo: 'baz' }]}
        columns={[
          {
            title: 'Testy test',
            dataIndex: 'foo',
          },
        ]}
        pagination={{
          next: 'asdfasdf',
          previous: 'mcnmcvmn',
          page_size: 50,
          count: 2,
          current_page_number: 1,
          total_pages: 3,
          onChange: jest.fn(),
        }}
      />
    );
    expect(component.container).toMatchSnapshot();
  });
});
