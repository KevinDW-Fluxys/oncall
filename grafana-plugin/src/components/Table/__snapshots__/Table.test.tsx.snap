// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table it renders pagination related data properly 1`] = `
<div>
  <div
    class="css-ukue92-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-bxa289-layoutChildrenWrapper"
    >
      <div
        class="rc-table root"
      >
        <div
          class="rc-table-container"
        >
          <div
            class="rc-table-content"
          >
            <table
              style="table-layout: auto;"
            >
              <colgroup />
              <thead
                class="rc-table-thead"
              >
                <tr>
                  <th
                    class="rc-table-cell"
                    scope="col"
                  >
                    Testy test
                  </th>
                </tr>
              </thead>
              <tbody
                class="rc-table-tbody"
              >
                <tr
                  class="rc-table-row rc-table-row-level-0 row-even"
                >
                  <td
                    class="rc-table-cell"
                  >
                    bar
                  </td>
                </tr>
                <tr
                  class="rc-table-row rc-table-row-level-0 row-odd"
                >
                  <td
                    class="rc-table-cell"
                  >
                    baz
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-bxa289-layoutChildrenWrapper"
    >
      <div
        class="pagination"
      >
        <div
          class="css-8f33us"
        >
          <ol>
            <li
              class="css-ds7o0y"
            >
              <button
                aria-label="previous"
                class="css-tb7nop-button"
                disabled=""
                type="button"
              >
                <span
                  class="css-1mhnkuh"
                >
                  <div
                    class="css-wf08df-Icon"
                  >
                    <svg
                      class="css-eyx4do"
                      height="16"
                      viewBox="0 0 24 24"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11.29,12l3.54-3.54a1,1,0,0,0,0-1.41,1,1,0,0,0-1.42,0L9.17,11.29a1,1,0,0,0,0,1.42L13.41,17a1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29,1,1,0,0,0,0-1.41Z"
                      />
                    </svg>
                  </div>
                </span>
              </button>
            </li>
            <li
              class="css-ds7o0y"
            >
              <button
                class="css-1pq88ji-button"
                type="button"
              >
                <span
                  class="css-1mhnkuh"
                >
                  1
                </span>
              </button>
            </li>
            <li
              class="css-ds7o0y"
            >
              <button
                class="css-tb7nop-button"
                type="button"
              >
                <span
                  class="css-1mhnkuh"
                >
                  2
                </span>
              </button>
            </li>
            <li
              class="css-ds7o0y"
            >
              <button
                class="css-tb7nop-button"
                type="button"
              >
                <span
                  class="css-1mhnkuh"
                >
                  3
                </span>
              </button>
            </li>
            <li
              class="css-ds7o0y"
            >
              <button
                aria-label="next"
                class="css-tb7nop-button"
                type="button"
              >
                <span
                  class="css-1mhnkuh"
                >
                  <div
                    class="css-wf08df-Icon"
                  >
                    <svg
                      class="css-eyx4do"
                      height="16"
                      viewBox="0 0 24 24"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
                      />
                    </svg>
                  </div>
                </span>
              </button>
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
</div>
`;
